features/step_definitions/user_steps.rb

```diff
@@ -541,4 +541,8 @@ def page_heading_text(page)
 
 Given('they continue to the idp') do
   click_on('continue-to-idp-button')
+end
+
+Given(/^the IDP fails to authenticate user$/) do
```

Comment:
What's the `/^` and `$/`? Is that regex? Do we need regex here? (we don't use regex anywhere else for step names)